### PR TITLE
[Bug] Add regular column when materialized slot is empty in tuple

### LINKED ISSUE #4716: Failed to query constant which is in inline view
**Describe the bug**

Situation: Select a constant column from inline view which includes at least one regular column.
Query failed!

```
MySQL [test]> select a from (select 'c1' a , k1 from test) a;
ERROR 1064 (HY000): errCode = 2, detailMessage = failed to build storage scanner, no materialized slot!
```

**To Reproduce**
1. create table 
```
CREATE TABLE `test` (
  `k1` int(11) NULL COMMENT "",
  `k2` bigint(20) NULL COMMENT "",
  `k3` decimal(9, 0) NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`k1`, `k2`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k1`) BUCKETS 10
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
```

2. insert values

```
insert into test values(2,2,2);
```

3. query

```
MySQL [test]> select a from (select 'c1' a , k1 from test) a;
ERROR 1064 (HY000): errCode = 2, detailMessage = failed to build storage scanner, no materialized slot!
```

**Expected behavior**

```
+------+
| a    |
+------+
| c1   |
+------+
```

If there are two lines in table, the result will be:

```
+------+
| a    |
+------+
| c1   |
| c1   |
+------+
```

### PR DESCRIPTION
**Pull request by EmmyMiao87**
If the materialized slot is empty in inline view, the scanner will throw exception 'no materialized slot!'.
When the outer query does not need any real columns of the inner query at all, such as the outer query only contains constant calculations,
all slots in the inner query will not be materialized.
For example:
```
select c1 from (select 'xx' c1, k1 from test) t1;
```

But this does not mean that the inner query (select 'xx' c1, k1 from test) does not need to be executed.
Because the number of rows in the outer result needs to be determined by the number of rows in the inner query.
In this case, the inner query scan node needs to add a column to ensure the correct result.
At the same time it can avoid the scan node reporting errors.

Fixed #4716

Change-Id: I86bb52efab65b658ea78094037543cb23dbff1d4

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kangkaisen** [comment]
+1. But please add a UT for this bug.

**Comment by EmmyMiao87** [comment]
> +1. But please add a UT for this bug.

This single test is a bit difficult to add, because the slot information is not displayed in the explain.

Or should I just add a single test for this function `materializeBaseTableRefResultForCrossJoinOrCountStar` ?

**Comment by kangkaisen** [comment]
> > +1. But please add a UT for this bug.
> 
> This single test is a bit difficult to add, because the slot information is not displayed in the explain.
> 
> Or should I just add a single test for this function `materializeBaseTableRefResultForCrossJoinOrCountStar` ?

OK

**Review by kangkaisen — DISMISSED** [review]
+1

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 1]
```diff
@@ -1902,7 +1902,7 @@ private void materializeTableResultForCrossJoinOrCountStar(TableRef tblRef, Anal
      * @param analyzer
      */
     private void materializeBaseTableRefResultForCrossJoinOrCountStar(BaseTableRef tblRef, Analyzer analyzer) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:1904]: I think this method name should be changed to a more common name?
like: `materializeSlotForEmptyMaterializedTableRef()`?
And add examples in comment to explain why we call it.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**